consensus/istanbul/ibft/core/handler.go

```diff
@@ -19,18 +19,20 @@ package core
 import (
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/consensus/istanbul"
+	istanbulcommon "github.com/ethereum/go-ethereum/consensus/istanbul/common"
+	ibfttypes "github.com/ethereum/go-ethereum/consensus/istanbul/ibft/types"
 )
 
 // Start implements core.Engine.Start
 func (c *core) Start() error {
-	// Start a new round from last sequence + 1
-	c.startNewRound(common.Big0)
-
 	// Tests will handle events itself, so we have to make subscribeEvents()
 	// be able to call in test.
 	c.subscribeEvents()
 	go c.handleEvents()
 
+	// Start a new round from last sequence + 1
+	c.startNewRound(common.Big0)
```

Comment:
why not like the line in final_committed.go:24?
```suggestion
	go c.startNewRound(common.Big0)
```

Reply:
`c.startNewRound(common.Big0)` in `consensus/istanbul/ibft/core/handler.go` is called when the Consensus Engine is first started and doesn't need to called in a goroutine, where as in `final_commited.go` it is called in a goroutine because during network transition, the legacy IBFT network is first stopped before starting QBFT consensus. As part of stopping legacy IBFT network, the engine is stopped as well and all the events are unsubscribed. Stopping of the engine is blocked until core/handler.go::handleEvents() is exited. The goroutine above makes sure handleEvents() exits due to unsubscribing of events and legacy IBFT engine is stopped

Reply:
Resolving conversation.

If updates are required we will cover them as part of the migration work.